docs/key-issues/cross_platform.md

```diff
@@ -0,0 +1,92 @@
+supported cross-compilation; however, these cross-compilation capabilities are
+easy to break unless they are exercised regularly.
+
+CPython's build system includes some support for cross-compilation. This support
+is largely based on leveraging autoconf's support for cross compilation. This
+support wasn't well integrated into distutils and the compilation of the binary
+portions of stdlib; however, with the deprecation and removal of disutils in
+Python 3.12, this situation has improved.
+
+The specification of PEP517 means cross-platform compilation support has been
+largely converted into a concern for individual build systems to manage.
+
+## Problems
+
+There is currently a small gap in communicating target platform details to the
+build system. While a build system like autoconf or Cmake may support
+cross-platform compilation, and a project may be able to cross-compile binary
+artefacts, invocation of the PEP517 build interface currently assumes that the
+platform running the build will be the platform that ultimately runs the Python
+code. As a result, `sys.platform`, or the various attributes of the `platform`
+library can't be used as part of the build process.
+
+`pip` provides limited support for installing binaries for a different platform
+by specifying a `--platform`, `--implementation` and `--abi` flags; however,
+these flags only work for the selection of pre-built binary artefacts.
+
+## History
+
+Tools like [crossenv](https://github.com/benfogle/crossenv) can be used to trick
+Python into performing cross-platform builds. These tools use path hacks and
+overrides of known sources of platform-specific details (like `distutils`) to
+provide a cross-compilation environment. However, these solutions tend to be
+somewhat fragile as they aren't first-class citizens of the Python ecosystem.
+
+[The BeeWare Project](https://beeware.org) also uses a version of these
+techniques. On both platforms, BeeWare provides a custom package index that
+contains pre-compiled binaries ([Android](https://chaquo.com/pypi-7.0/);
+[iOS](https://anaconda.org/beeware/repo)). These binaries are produced using a
+forge-like set of tooling
+([Android](https://github.com/chaquo/chaquopy/tree/master/server/pypi);
+[iOS](https://github.com/freakboy3742/chaquopy/tree/iOS-support/server/pypi)).
+
+## Relevant resources
+
+TODO
+
+## Potential solutions or mitigations
+
+At it's core, what is required is a recognition that cross-platform builds as a
+use case that the Python ecosystem supports.
+
+In concrete terms, for native modules, this would require either:
+
+1. Extension of the PEP517 interface to allow communicating the desired target
+   platform as part of a binary build; or
+
+2. Formalization of the "platform identification" interface that can used by
+   PEP517 build backends to identify the target platform, so that tools like
+   `crossenv` can provide a reliable proxied environment for cross-platform
+   builds.
```

Comment:
I think in terms of requirements I'd add (resp. subsume):
3. metadata separation into build & target requirements
4. infrastructure support (e.g. hooks, etc.) to tell the PEP517 interface that certain steps (e.g. code generation) cannot be run on the target hardware